packages/curve-ui-kit/src/utils/time.utils.ts

```diff
@@ -24,3 +28,26 @@ export const waitFor = async (
   }
   await handleTimeout<void>(waitUntil(), timeout, message)
 }
+
+/**
+ * Waits for approval by checking isApproved, and if not approved, calls onApprove,
+ * waits for the transactions to be mined and then waits until isApproved returns true.
+ */
+export async function waitForApproval({
```

Comment:
Could argue to move this to a `wagmi.ts` util function, or the existing `web3.ts`. I know there's a timeout, but the primary theme here are approvals, not the timing itself. Not too important.